homeassistant/components/homekit/lights.py

```diff
@@ -0,0 +1,72 @@
+"""Class to hold all light accesories."""
+import logging
+
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.helpers.event import async_track_state_change
+
+from . import TYPES
+from .accessories import HomeAccessory
+from .const import (
+    SERVICES_LIGHTBULB, CHAR_BRIGHTNESS, CHAR_STATE)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@TYPES.register('Lightbulb')
+class Lightbulb(HomeAccessory):
+    """Generate a Light accessory for a lightbulb."""
+
+    def __init__(self, hass, entity_id, display_name):
+        _LOGGER.debug("Loading light component1")
+        """Initialize a LightBulb accessory object."""
+        super().__init__(display_name)
+        self.set_category(self.ALL_CATEGORIES.LIGHTBULB)
+        _LOGGER.debug("Loading light component2")
+        self.set_accessory_info(entity_id)
+        _LOGGER.debug("Loading light component3")
+        self.add_preload_service(SERVICES_LIGHTBULB)
```

Comment:
To save you some headache later. It seems that `add_preload_service` doesn't work if you plan to add an additional characteristic to the service. This has to be done before the service is added to the accessory. I don't know the reason to it jet. This could work:
```py
from pyhap.loader import get_serv_loader
from pyhap.loader import get_char_loader
serv = get_serv_loader().get(SERVICES_LIGHTBULB)
brightness = get_char_loader().get(CHAR_BRIGHTNESS)
serv.add_opt_characteristic(brightness)
self.add_service(serv)
```

Reply:
So that's why that wasn't working.. Thank you!